### PR TITLE
revamp/cleanup JSDoc + tests

### PR DESCRIPTION
**Pull request by Raynos**
There were some JSDoc warnings in the functions app,
making changes here in fake-lambda-api-gateway to
clean it up

 - update jsconfig
 - Add typescript back
 - Run typescript as part of lint/test
 - Change test suite from tape to tapzero
 - Change TestHarness pattern into TestCommon pattern
 - Include test/errors.js in npm test
 - Port test/errors.js to TestCommon.
 - Add githook.
 - Fix lint / typescript errors

### CODE REVIEW

File: docker.js [hunk 1]
```diff
@@ -141,7 +144,7 @@ class DockerLambda {
     await dockerLambdaReady(this.port)
   }
 
-  async request (id, eventObject) {
+  async request (_id, eventObject) {
```
Comment (jwerle) [inline docker.js:147]: what the `_` here? because it is unused?

File: index.js [hunk 2]
```diff
@@ -481,3 +495,33 @@ function cuuid () {
   const str = (Date.now().toString(16) + Math.random().toString(16).slice(2) + Math.random().toString(16).slice(2) + Math.random().toString(16).slice(2)).slice(0, 32)
```
Comment (jwerle) [inline index.js:495]: wow

File: docker.js (hunk 1 above)
Comment (Raynos) [reply docker.js:147]: linter / typescript complained about unused function parameter so i renamed it to `_id` to get it to shut up

File: index.js (hunk 2 above)
Comment (Raynos) [reply index.js:495]: this function is a few hundreds of lines shorter the uuid package.

**Review by jwerle — APPROVED** [review]

File: index.js (hunk 2 above)
Comment (jwerle) [reply index.js:495]: 😂

File: docker.js (hunk 1 above)
Comment (jwerle) [reply docker.js:147]: good to know!

**Pull request merged**